lib/saml_authenticator.rb

```diff
@@ -62,6 +61,18 @@ def after_authenticate(auth)
       ::PluginStore.set("saml", "saml_user_#{uid}", {user_id: result.user.id })
     end
 
+    if GlobalSetting.try(:saml_validate_email_fields).present? && auth.extra[:raw_info].attributes['memberOf'].present?
+      unless (GlobalSetting.try(:saml_validate_email_fields).split("|").map(&:downcase) & auth.extra[:raw_info].attributes['memberOf'].map(&:downcase)).empty?
+        result.email_valid = true
+      else
+        result.email_valid = false
+      end
+    elsif GlobalSetting.try(:saml_default_emails_valid).present? && GlobalSetting.try(:saml_default_emails_valid) == "false"
```

Comment:
" GlobalSetting.try(:saml_default_emails_valid) == "false"" you 100% sure about this ... GlobalSetting should coerce to a boolean ... GlobalSetting.try(:saml_default_emails_valid) so in theory present? would only be "true" if it is actually set to try... try it out.

Reply:
You were right. I made the required changes here.

Can you do a quick review again? Not urgent at all.